### ISSUE TITLE
sql: support correlated subquery in DELETE WHERE clause

### ISSUE THREAD

**Issue by knz** [cockroachdb/cockroach#31670]
**Is your feature request related to a problem? Please describe.**

Found via the Hibernate test org.hibernate.test.hql.BulkManipulationTest.testDeleteWithSubquery

```sql
        delete
        from
            MANY_TO_MANY
        where
            (
                IN_ID
            ) in (
                select
                    ID
                from
                    SIMPLE_1
                where
                    (
                        select
                            count(associated1_.SIMPLE_1_ID)
                        from
                            SIMPLE_2 associated1_
                        where
                            SIMPLE_1.ID=associated1_.SIMPLE_1_ID
                    )=0
                    and (
                        NAME like '%'
                    )
                )
```

**Comment by knz** [comment]
Another:

```sql
        delete
        from
            Attrvalue
        where
            id in (
                select
                    attrvalue4_.id
                from
                    Employee employee1_ cross
                join
                    Employeegroup employeegr2_ cross
                join
                    Attrset attrset3_ cross
                join
                    Attrvalue attrvalue4_
                where
                    employeegr2_.id=employee1_.employeegroup_id
                    and attrset3_.id=employee1_.attrset_id
                    and (
                        attrvalue4_.id in (
                            select
                                attrvalues5_.attrvalues_id
                            from
                                ATTRSET_X_ATTRVALUE attrvalues5_
                            where
                                attrset3_.id=attrvalues5_.Attrset_id
                        )
                    )
                )
```

**Comment by awoods187** [comment]
@andy-kimball would this work once mutations are supported in 2.2? Would this need the apply operator @jordanlewis ?

**Comment by knz** [comment]
Yes optimizer knowledge of mutation will be 70% of the work needed here. For the requirement on apply, it will depend on the query I think. Not sure about those quoted above.

**Comment by andy-kimball** [comment]
Yes, I'd expect these to work once we added CRUD support to the optimizer. They shouldn't need `Apply`.

**Comment by BramGruneir** [comment]
Closed thanks to #33524.
